Library/src/CSPFoundation.cpp

```diff
@@ -255,10 +255,15 @@ bool CSPFoundation::Initialise(const csp::common::String& EndpointRootURI, const
 	const std::string UserServiceURI		= RootURI + "/mag-user";
 	const std::string PrototypeServiceURI	= RootURI + "/mag-prototype";
 	const std::string SpatialDataServiceURI = RootURI + "/mag-spatialdata";
-	const std::string MultiplayerServiceURI = RootURI + "/mag-multiplayer/hubs/v1/multiplayer";
 	const std::string AggregationServiceURI = RootURI + "/oly-aggregation";
 	const std::string TrackingServiceURI	= RootURI + "/mag-tracking";
 
+	const std::string Hostname = "ogs", MultiplayerHostname = Hostname + "-multiplayer";
+	const std::string RootMultiplayerURI
+		= RootURI.substr(0, RootURI.find(Hostname)) + MultiplayerHostname + RootURI.substr(RootURI.find(Hostname) + Hostname.length());
+	const std::string MultiplayerServiceURI = RootMultiplayerURI + "/mag-multiplayer/hubs/v1/multiplayer";
```

Comment:
[Consider] It's pretty hard to follow what this is doing without understanding what the intended constructed string is, a comment or two describing the string transformation could help.
I wonder if it's necessary to go to the effort to transform this from the `RootURI`, could we not directly encode the new multiplayer URI and void some of this transformation?

Reply:
Sure I'll add comments. 

I constructed it from RootURI because it is not always the same. It is passed as an argument in the Foundation initialisation. The production one is ogs.magnopus-something, but there are also others like ogs-internal.magnopus-something. 
I made the assumption that ogs-multiplayer-internal was the right order, thinking if it wasn't the CI would tell me, but now I am not so sure we have the test coverage for that.

I reached out to CHS and the person to ask was off yesterday, back on Monday

Reply:
Ah right, sorry, I had thought this was in the tests, not the core lib itself.
Aye ... it is rather difficult to be certain that this is fully correct. I couldn't say for sure if this URI construction was always right, or is just right in the case of the particular services CHS has stood up right now.

Reply:
Agreed, I'll ask that person on Monday for guidance

Reply:
Comment request fixed in 18eb922b567bae6e56fc48f7b657eb7cf5257a3d